.github/workflows/pull-integration-release.yaml

```diff
@@ -19,25 +19,46 @@ on:
       - 'release-**'
 
 jobs:
-  wait-for-image-build:
-    environment: ${{ github.event.pull_request.author_association != 'COLLABORATOR' && github.event.pull_request.author_association != 'OWNER' && 'restricted' || 'internal' }}
-    name: Wait for image build
+  filter-changes:
+    name: List relevant VCS changes
+    outputs:
+      check: ${{ steps.changed-files.outputs.any_modified }}
     runs-on: ubuntu-latest
-    if: github.event.pull_request.draft == false
     steps:
       - uses: actions/checkout@v4
+      - uses: tj-actions/changed-files@d6babd6899969df1a11d14c368283ea4436bca78
+        id: changed-files
         with:
-          fetch-depth: 0
-      - uses: ./.github/actions/wait-for-job-succeed-or-fail
+          files_ignore: |
+            docs/**
+            **/*.md
+            OWNERS
+            CODEOWNERS
+            sec-scanners-config.yaml
+            .github/**
+            .**
+      - name: List all changed files
+        run: echo '${{ steps.changed-files.outputs.all_changed_files }}'
+
+  build-image:
+    name: Build and push manager image
+    needs: [filter-changes]
+    environment: ${{ github.event.pull_request.author_association != 'COLLABORATOR' && github.event.pull_request.author_association != 'OWNER' && 'restricted' || 'internal' }}
+    runs-on: ubuntu-latest
+    if: ${{ github.event.pull_request.draft == false && needs.filter-changes.outputs.check == 'true' }}
+    steps:
+      - uses: actions/checkout@v4
```

Comment:
There is a fix in the Istio workflow, that would be even more relevant for this, since in this workflow the image is used for testing: https://github.com/kyma-project/istio/pull/943